src/main.rs

```diff
@@ -200,24 +226,25 @@ async fn update_receipt_status(
 }
 
 async fn handle_outcomes(
-    outcomes: &[near_indexer::Outcome],
+    outcomes: &HashMap<
+        near_indexer::near_primitives::hash::CryptoHash,
+        near_indexer::near_primitives::views::ExecutionOutcomeWithIdView,
+    >,
     pool: &Pool<ConnectionManager<PgConnection>>,
 ) {
     let mut failed_receipt_ids: Vec<String> = vec![];
     let mut succeeded_receipt_ids: Vec<String> = vec![];
 
-    for outcome in outcomes {
-        if let near_indexer::Outcome::Receipt(execution_outcome) = outcome {
-            match execution_outcome.outcome.status {
-                near_indexer::near_primitives::views::ExecutionStatusView::Failure(_) => {
-                    failed_receipt_ids.push(execution_outcome.id.to_string())
-                }
-                near_indexer::near_primitives::views::ExecutionStatusView::SuccessReceiptId(_)
-                | near_indexer::near_primitives::views::ExecutionStatusView::SuccessValue(_) => {
-                    succeeded_receipt_ids.push(execution_outcome.id.to_string());
-                }
-                _ => {}
+    for outcome in outcomes.values() {
+        match outcome.outcome.status {
+            near_indexer::near_primitives::views::ExecutionStatusView::Failure(_) => {
+                failed_receipt_ids.push(outcome.id.to_string())
+            }
+            near_indexer::near_primitives::views::ExecutionStatusView::SuccessReceiptId(_)
+            | near_indexer::near_primitives::views::ExecutionStatusView::SuccessValue(_) => {
+                succeeded_receipt_ids.push(outcome.id.to_string());
             }
+            _ => {}
```

Comment:
Let's explicitly match against the only left variant `Unknown` and report an error here as we don't expect it.